reason-react-native/src/apis/AppRegistry.md

```diff
@@ -5,14 +5,103 @@ wip: true
 ---
 
 ```reason
+type task('data) = 'data => Js.Promise.t(unit);
+type taskProvider('data) = unit => task('data);
+type taskCanceller = unit => unit;
+type taskCancelProvider = unit => taskCanceller;
+
+type componentProvider('a) = unit => React.component('a);
+type wrapperComponentProvider('a, 'b) = 'b => React.component('a);
+
+type appParameters;
+
+external asAppParameters: 'a => appParameters = "%identity";
+
+type appConfig;
+
+[@bs.obj]
+external appConfig:
+  (
+    ~appKey: string,
+    ~component: componentProvider('a)=?,
+    ~run: appParameters => unit=?,
+    ~section: bool=?,
+    unit
+  ) =>
+  appConfig =
+  "";
+
+type runnable('a) = {
+  .
+  "component": Js.Nullable.t(componentProvider('a)),
+  "run": appParameters => unit,
+};
+
+type registry('a) = {
+  .
+  "sections": array(string),
+  "runnables": Js.Dict.t(runnable('a)),
+};
+
+[@bs.module "react-native"] [@bs.scope "AppRegistry"]
+external getAppKeys: unit => array(string) = "";
+
+[@bs.module "react-native"] [@bs.scope "AppRegistry"]
+external getRegistry: unit => registry('a) = "";
+
+[@bs.module "react-native"] [@bs.scope "AppRegistry"]
+external getRunnable: string => Js.Nullable.t(runnable('a)) = "";
```

Comment:
I think this should only return `undefined`, not `null`, so we could use `option` instead of `Js.Nullable.t`.

Reply:
You're right. I was probably too conservative, mapping `?` to `undefined | null` by default, but since this is looking up a value in a hash map, `option` should be enough.